TransformDragger.rbxmx

```diff
@@ -6624,5 +6616,88 @@ return module
 ]]></ProtectedString>
 			</Properties>
 		</Item>
+		<Item class="ModuleScript" referent="RBXEC7C6085619E450693F32FA161266FC2">
+			<Properties>
+				<Content name="LinkedSource"><null></null></Content>
+				<string name="Name">RecyclingBin</string>
+				<string name="ScriptGuid"></string>
+				<ProtectedString name="Source"><![CDATA[-- @CloneTrooper1019 <3
+
+local recyclingBin = {}
+local bin = {}
+local expirationTime = 5
+
+function recyclingBin:Recycle(object)
+	local class = object.ClassName
+	if not bin[class] then
+		bin[class] = {}
+	end
+	bin[class][object] = tick()+expirationTime;
+	object.Parent = nil
+end
+
+function recyclingBin:RecycleObjects(objects)
+	for _,v in pairs(objects) do
+		self:Recycle(v)
+	end	
+end
+
+function recyclingBin:Pull(objectType,parent)
+	local object do
+		local objects = bin[objectType]
+		if objects then
+			object = next(objects)
+			if object then
+				objects[object] = nil
+			end
+		end
+	end
+	if not object then
+		object = Instance.new(objectType)
+	end
+	if parent then
+		object.Parent = parent
+	end
+	return object
+end
+
+-- Allocate works similarly to Pull, except it tries to reuse objects that exist in the specified bin.
+-- This removes the need to recycle every object during every frame.
+
+function recyclingBin:Allocate(objectType,needed,bin)
+	local allocation = {}
+	local allocated = 0
+	for _,v in pairs(bin:GetChildren()) do
+		if v:IsA(objectType) then
+			allocated = allocated + 1
+			if allocated > needed then
+				self:Recycle(v)
+			else
+				allocation[allocated] = v
+			end
+		end
+	end
+	while allocated < needed do
+		allocated = allocated + 1
+		allocation[allocated] = self:Pull(objectType,bin)
+	end
+	return allocation
+end
+
+spawn(function ()
+	while wait(1) do
+		for _,queue in pairs(bin) do
+			for object,expireTime in pairs(queue) do
```

Comment:
This isn't too much of an issue but it seems like iterating through the entire queue table every second could be avoided by maintaining the queue in a different way. If you use a queue data structure with fast addition and removal from either end of the queue then you can just check if the top of the queue has an expiration time less than the current tick.